lib/Utilities/IsArangoExecutable.h

```diff
@@ -0,0 +1,31 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef ARANGODB_UTILITIES_IS_EXECUTABLE_H
+#define ARANGODB_UTILITIES_IS_EXECUTABLE_H 1
+
+//////////////////////////////////////////////////////////////////////////////
+/// @brief tell whether str contains a string matching one of our executables
+//////////////////////////////////////////////////////////////////////////////
+bool HasShellExecutableNameInIt(std::string& str);
```

Comment:
The function signature looks a bit strange, as one would expect a function with name "Has...(...)" to be free of side-effects.
But this function can modify its input value, so it's not free of side-effects.
I suggest changing it to something like `std::string extractShellExecutableName(std::string const& input);`, which won't modify its input, and return the name of the executable found by value. If the return value is empty, this means "no shell executable used".